tests/v2/validation/provisioning/rke1/provisioning_node_driver_test.go

```diff
@@ -115,68 +114,45 @@ func (r *RKE1NodeDriverProvisioningTestSuite) ProvisioningRKE1Cluster(provider P
 		{"3 nodes - 1 role per node Standard User", nodeRoles1, r.standardUserClient},
 	}
 
-	var name string
+	var name, scaleName string
 	for _, tt := range tests {
 		subSession := r.session.NewSession()
 		defer subSession.Cleanup()
 
 		client, err := tt.client.WithSession(subSession)
 		require.NoError(r.T(), err)
 
-		for _, kubeVersion := range r.kubernetesVersions {
-			name = tt.name + providerName + " Kubernetes version: " + kubeVersion
-			for _, cni := range r.cnis {
-				name += " cni: " + cni
-				r.Run(name, func() {
-					testSession := session.NewSession(r.T())
-					defer testSession.Cleanup()
-
-					testSessionClient, err := tt.client.WithSession(testSession)
-					require.NoError(r.T(), err)
-
-					clusterName := namegen.AppendRandomString(provider.Name)
-
-					cluster := clusters.NewRKE1ClusterConfig(clusterName, cni, kubeVersion, testSessionClient)
-
-					clusterResp, err := clusters.CreateRKE1Cluster(testSessionClient, cluster)
-					require.NoError(r.T(), err)
-
-					nodeTemplateResp, err := provider.NodeTemplateFunc(client)
-					require.NoError(r.T(), err)
-
-					nodePool, err := nodepools.NodePoolSetup(testSessionClient, tt.nodeRoles, clusterResp.ID, nodeTemplateResp.ID)
-					require.NoError(r.T(), err)
-
-					nodePoolName := nodePool.Name
-
-					opts := metav1.ListOptions{
-						FieldSelector:  "metadata.name=" + clusterResp.ID,
-						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
-					}
-					watchInterface, err := r.client.GetManagementWatchInterface(management.ClusterType, opts)
-					require.NoError(r.T(), err)
-
-					checkFunc := clusters.IsHostedProvisioningClusterReady
-
-					err = wait.WatchWait(watchInterface, checkFunc)
-					require.NoError(r.T(), err)
-					assert.Equal(r.T(), clusterName, clusterResp.Name)
-					assert.Equal(r.T(), nodePoolName, nodePool.Name)
-
-					clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
-					require.NoError(r.T(), err)
-					assert.NotEmpty(r.T(), clusterToken)
-
-					err = nodepools.ScaleWorkerNodePool(testSessionClient, tt.nodeRoles, clusterResp.ID, nodeTemplateResp.ID)
-					require.NoError(r.T(), err)
-				})
+		for _, providerName := range r.providers {
+			provider := CreateProvider(providerName)
+			providerName := " Node Provider: " + provider.Name
+			for _, kubeVersion := range r.kubernetesVersions {
+				name = tt.name + providerName + " Kubernetes version: " + kubeVersion
+				scaleName = tt.name + providerName + " Kubernetes version: " + kubeVersion
```

Comment:
One little nitpick, you can don't need `scaleName` here. At the bottom just do `scaleName += name + " scaling"`

Reply:
Good catch! Resolved it